### PR TITLE
Fix CVEs and update dependencies

### PR DESCRIPTION
**Pull request by sss-ryun**
**Description:** Fixes CVE-2021-42550, CVE-2023-34462, CVE-2023-44487, and CVE-2023-34062. Does not fix, however, Cxeb68d52e-5509, CVE-2021-37714, CVE-2020-13956, and CVE-2021-29425. These other ones need to be fixed upstream in lavaplayer.

**Justification:** I don't feel comfortable nor should anyone, using Discord4J as long as it has CVEs. These CVEs might not affect Discord4J, but it is better to side with caution.

This version builds and tested using `./gradlew test`.

Additionally, old dependencies were updated because Discord4J uses 2-3 years old versions.

### CODE REVIEW

**Comment by sss-ryun** [comment]
I currently have this published locally and am using it to build my first Discord4J bot

**Comment by Doc94** [comment]
Looks like this updates breaks the compile with Java 8

**Comment by sss-ryun** [comment]
> Looks like this updates breaks the compile with Java 8

Strange. This compiles for me locally. I've been using it all day to transfer my old bot from JS to Kotlin. It should be fixable

**Comment by sss-ryun** [comment]
I'm using Coretto 1.8

File: build.gradle [hunk 1]
```diff
@@ -24,17 +24,17 @@ ext {
...
+    caffeine_version = '2.9.3'
     immutables_group = 'org.immutables'
-    immutables_version = '2.9.0'
+    immutables_version = '2.10.1'
 
     // Test dependencies
     junit_version = '5.7.1'
     mockito_version = '3.11.2'
-    logback_version = '1.2.3'
+    logback_version = '1.5.6'
```
Comment (Azn9) [inline build.gradle:37]: Only logback versions < 1.4.X are compatible with java 8 (https://logback.qos.ch/dependencies.html)

Comment (Azn9) [reply build.gradle:37]: /cc @Doc94

**Review by Azn9 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (sss-ryun) [reply build.gradle]: I have a fix up. I found it in their Github [README.md](https://github.com/qos-ch/logback/blob/master/README.md)

> The 1.3.x series requires Java 8 at runtime. If you wish to build logback from source, you will need Java 9.

**Comment by sss-ryun** [comment]
This one should be fixed for all CVEs in logback

**Comment by sss-ryun** [comment]
However, lavaplayer still needs to be fixed upstream

**Comment by Azn9** [comment]
This should target 3.2.X, I'll rebase it

**Review by Azn9 — APPROVED** [review]

File: common/build.gradle [hunk 2]
```diff
@@ -1,8 +1,8 @@
 apply plugin: 'com.gorylenko.gradle-git-properties'
 
 dependencies {
-    api "io.projectreactor.netty:reactor-netty-http"
-    api "io.projectreactor.addons:reactor-extra"
+    api "io.projectreactor.netty:reactor-netty-http:1.1.18"
+    api "io.projectreactor.addons:reactor-extra:3.5.1"
```
Comment (Doc94) [inline common/build.gradle:5]: i feel this need to be moved to a var like the rest of version definitions.

**Review by Doc94 — CHANGES_REQUESTED** [review]

**Pull request merged**